docs/manual/howto/git.rst

```diff
@@ -0,0 +1,158 @@
+tests locally helps you find stupid programming mistakes like typos.
+
+Now it is time to add your work to the *staging area* and *commit* your changes.
+``git status`` tells you which files you have created or modified. With all the
+files you need for your feature, do::
+
+  git add path/to/file1 path/to/file2 ...
+
+These files are now in the staging area.
+
+.. hint:: If you changed a file and want to select specific changes and want to
+   keep others outside the staging area, you can use ``git add -p``.
+
+Once your changes are selected and ready, you can commit them::
+
+  git commit
+
+This will open a text editor with a few lines that have been commented out to
+once again show you a summary of what you did. It it time to write a commit
+message. The first line is a summary, like the subject of an email and should be
+short. This summary is followed by a blank line and then a more detailed
+description of your changes. Your commit message should comply to the
+`standard format <https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html>`_
+and our :ref:`contribution guidelines <submitting-a-pr>`.
+
+Once you have committed all your changes and are ready to create the pull
+request, push your changes to your fork on GitHub::
+
+  git push -u your_username feature-branch
+
+If you want to push more commits later, you can just use
+``git push your_username``. Once that is done, you can visit GitHub, navigate to
+your fork or our repository and you will be greeted with a box saying that your
+branch recently had pushes to it. Click on the appropriate button to create a
+pull request. If that box does not show up, go to our repository, click on pull
+requests and *new pull request*.
+
+I committed changes and the tests failed
+========================================
+
+You can easily change your last commit: After you have done your work,
+``git add`` everything you need and use ``git commit --amend`` to change your
+last commit. This causes the git history of your local clone to be diverged from
+your fork on GitHub, so you need to force-push your changes with::
+
+  git push -f your_username
+
+I was told to rebase my work
+============================
+
+If *upstream/master* is changed and you happened to change the same files as the
+commits that were added upstream, you should rebase your work onto the most
+recent *upstream/master*. Checkout your master, pull from *upstream*, checkout
+your branch again and then rebase it::
+
+  git checkout master
+  git pull upstream/master
+  git checkout feature-branch
+  git rebase upstream/master
+
+You will be asked to solve conflicts where your diff cannot be applied with
+confidence to the work that was pushed upstream. If that is the case, open the
+files in your text editor and resolve the conflicts manually. You possibly need
+to ``git rebase --continue`` after you have resolved conflicts for one commit if
+you are rebasing multiple commits.
```

Comment:
Add a note that rebasing is harder when working off master without a feature branch?

Reply:
Nah you already say to use a feature branch elsewhere, should be fine